### ISSUE TITLE
Use of $FILE() and $PIPE() together causes exception

### ISSUE THREAD

**Issue by jpmens** [ansible/ansible#1118]
Mentioned on IRC late last night:

```
ansible 0.8 (devel 69517832cd) last updated 2012/09/26 07:58:55 (GMT +200)
```
## Playbook

``` yaml

---
- hosts: 127.0.0.1
  gather_facts: false
  connection: local
  vars:
  - person: Jane Jolie
  - pcont: $PIPE(/bin/date)
  - fcont: $FILE(/tmp/small.file)
  tasks:
  - action: command /bin/echo ${pcont}
  - action: command /bin/echo ${fcont}
#  - action: command /bin/echo ${fcont} and ${pcont}
#    action: template src=t1.j2 dest=/tmp/t1
```

If I uncomment the third action (i.e. use ${fcont} and ${pcont} together), I get the following exception:

```
Traceback (most recent call last):
  File "/Users/jpm/Auto/pubgit/ansible/ansible/bin/ansible-playbook", line 151, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/Users/jpm/Auto/pubgit/ansible/ansible/bin/ansible-playbook", line 127, in main
    pb.run()
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/__init__.py", line 161, in run
    play = Play(self, play_ds, play_basedir)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/play.py", line 83, in __init__
    self._tasks      = self._load_tasks(self._ds, 'tasks')
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/play.py", line 119, in _load_tasks
    results.append(Task(self,y,module_vars=mv))
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/task.py", line 112, in __init__
    self.name = utils.template(None, self.name, self.module_vars)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/utils.py", line 295, in template
    text = varReplaceFilesAndPipes(basedir, text)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/utils.py", line 268, in varReplaceFilesAndPipes
    raise VarNotFoundException()
ansible.utils.VarNotFoundException
```

Similarly, use of these variables together in a template fails. Uncomment the template action to get

```

TASK: [template src=t1.j2 dest=/tmp/t1] ********************* 
fatal: [127.0.0.1] => {'msg': '', 'failed': True}
fatal: [127.0.0.1] => {'msg': '', 'failed': True}
```

The template contains

``` jinja2

FILECONTENT=[{{ fcont }}]
PIPECONTENT=[${pcont}]
```

**Comment by mpdehaan** [comment]
Note to self VarNotFound example is also not acceptable -- we should subclass errors.AnsibleErrors and catch the traceback and provide a proper message

**Comment by dhozac** [comment]
#1149 should likely fix this, and #1147 fixes the exception.
